addons/pos_loyalty/models/pos_config.py

```diff
@@ -7,58 +7,18 @@
-
-    use_coupon_programs = fields.Boolean('Coupons & Promotions',
-        help="Use coupon and promotion programs in this PoS configuration.")
-    coupon_program_ids = fields.Many2many(
-        'loyalty.program', string="Coupon Programs", domain=[('program_type', '=', 'coupons')],
-        relation='pos_config_coupon_program_rel')
-    promo_program_ids = fields.Many2many(
-        'loyalty.program', string="Promotion Programs", domain=[('program_type', '=', 'promotion')],
-        relation='pos_config_promo_program_rel')
-
-    loyalty_program_id = fields.Many2one('loyalty.program', "PoS Loyalty Programs", domain=[('program_type', '=', 'loyalty')], default=_default_loyalty_program)
-
-    use_gift_card = fields.Boolean('Gift Cards')
-    gift_card_program_id = fields.Many2one('loyalty.program', "PoS Gift Card Program", domain=[('program_type', '=', 'gift_card')])
     gift_card_settings = fields.Selection(
         [
-            ("create_set", "Generate a new barcode and set a price"),
-            ("scan_set", "Scan an existing barcode and set a price"),
-            ("scan_use", "Scan an existing barcode with an existing price"),
+            ("create_set", "Generate PDF cards"),
+            ("scan_use", "Scan existing cards"),
         ],
         string="Gift Cards settings",
         default="create_set",
         help="Defines the way you want to set your gift cards.",
     )
 
     # While we select all program types separately they will all behave the same
-    all_program_ids = fields.Many2many('loyalty.program', compute='_compute_all_programs')
-
-    @api.depends('use_coupon_programs', 'coupon_program_ids', 'promo_program_ids',
-        'loyalty_program_id', 'use_gift_card', 'gift_card_program_id')
-    def _compute_all_programs(self):
-        for config in self:
-            programs = self.env['loyalty.program']
-            if config.use_coupon_programs:
-                programs |= config.coupon_program_ids
-                programs |= config.promo_program_ids
-            # This may be a separate field on the config but it actually will be handled just like any other program
-            if config.loyalty_program_id:
-                programs |= config.loyalty_program_id
-            # We also include the gift card program to be able to claim the reward (discount)
-            # This one will behave a little differently as it will display more options
-            if config.use_gift_card:
-                programs |= config.gift_card_program_id
-            config.all_program_ids = programs
-
-    @api.model
-    def set_loyalty_program_to_main_config(self):
-        main_config = self.env.ref('point_of_sale.pos_config_main')
-        default_loyalty_program = self._default_loyalty_program()
-        main_config.write({'module_pos_loyalty': bool(default_loyalty_program), 'loyalty_program_id': default_loyalty_program.id})
+    all_program_ids = fields.Many2many('loyalty.program')
```

Comment:
As this field is not store before, we should do a migration script that'll take the M2M values from 'coupon_program_ids', 'promo_program_ids', 'loyalty_program_id' and 'gift_card_program_id' and put them in the field 'all_program_ids'.

But if the PR pass without the migration script, we can merge without the script and I can check to it next monday.

FYI @trj-odoo